in_toto/gpg/common.py

```diff
@@ -0,0 +1,152 @@
+"""
+<Module Name>
+  common.py
+
+<Author>
+  Santiago Torres-Arias <santiago@nyu.edu>
+
+<Started>
+  Nov 15, 2017
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Provides algorithm-agnostic gpg public key and signature parsing functions.
+  The functions select the appropriate functions for each algorithm and
+  call them.
+
+"""
+import struct
+import binascii
+
+import in_toto.log
+import in_toto.gpg.util
+from in_toto.gpg.constants import (PACKET_TYPES,
+        SUPPORTED_PUBKEY_PACKET_VERSIONS, SIGNATURE_TYPE_CANONICAL,
+        SUPPORTED_SIGNATURE_PACKET_VERSIONS, SUPPORTED_SIGNATURE_ALGORITHMS,
+        SUPPORTED_HASH_ALGORITHMS, SIGNATURE_HANDLERS, FULL_KEYID_SUBPACKET,
+        FULLY_SUPPORTED_MIN_VERSION)
+
+
+# XXX this doesn't support armored pubkey packets, so use with care.
+# pubkey packets are a little bit more complicated than the signature ones
+def parse_pubkey_packet(data):
+
+  if not data:
+    raise ValueError("Could not parse empty pubkey packet.")
+
+  data = in_toto.gpg.util.parse_packet_header(
+      data, PACKET_TYPES['main_pubkey_packet'])
+
+  ptr = 0
+  keyinfo = {}
+  version_number = data[ptr]
+  ptr += 1
+  if version_number not in SUPPORTED_PUBKEY_PACKET_VERSIONS: # pragma: no cover
+    raise ValueError("This pubkey packet version is not supported!")
+
+  time_of_creation = struct.unpack(">I", data[ptr:ptr + 4])
```

Comment:
This never gets used.

Reply:
I left these values in case we wanted to have a `-v` option akin to

```bash
Signature created on: Mon Jan  8 12:47:57 EST 2018
Hashed subpackets:
    .....
```

I also don't know how much it hurts to deserialize them, as it's a data structure following that format. I could probably leave a comment describing this though (and probably group the unused values). What do you think?

Reply:
Comment sounds good